projects/vdk-plugins/vdk-impala/tests/functional/test_impala_lineage.py

```diff
@@ -0,0 +1,363 @@
+# Copyright 2021 VMware, Inc.
+# SPDX-License-Identifier: Apache-2.0
+import json
+import os
+from unittest import mock
+from unittest import TestCase
+from unittest.mock import patch
+
+import pytest
+from click.testing import Result
+from vdk.api.lineage.model.logger.lineage_logger import ILineageLogger
+from vdk.api.lineage.model.sql.model import LineageData
+from vdk.api.lineage.model.sql.model import LineageTable
+from vdk.api.plugin.hook_markers import hookimpl
+from vdk.plugin.impala import impala_plugin
+from vdk.plugin.impala.impala_plugin import LINEAGE_LOGGER_KEY
+from vdk.plugin.test_utils.util_funcs import cli_assert_equal
+from vdk.plugin.test_utils.util_funcs import CliEntryBasedTestRunner
+from vdk.plugin.test_utils.util_funcs import jobs_path_from_caller_directory
+
+VDK_DB_DEFAULT_TYPE = "VDK_DB_DEFAULT_TYPE"
+VDK_IMPALA_HOST = "VDK_IMPALA_HOST"
+VDK_IMPALA_PORT = "VDK_IMPALA_PORT"
+VDK_IMPALA_SYNC_DDL = "VDK_IMPALA_SYNC_DDL"
+VDK_IMPALA_QUERY_POOL = "VDK_IMPALA_QUERY_POOL"
+
+
+class TestConfigPlugin:
+    def __init__(self, lineage_logger: ILineageLogger):
+        self.lineage_logger = lineage_logger
+
+    @hookimpl
+    def vdk_initialize(self, context):
+        # register a lineage logger which will be used to validate if correct lineage data was emitted
+        context.state.set(LINEAGE_LOGGER_KEY, self.lineage_logger)
+
+
+def execute_query(runner, query: str):
+    result: Result = runner.invoke(["impala-query", "--query", query])
+    cli_assert_equal(0, result)
+
+
+@pytest.mark.usefixtures("impala_service")
+class ImpalaLineageTest(TestCase):
+    @patch.dict(
+        os.environ,
+        {
+            VDK_DB_DEFAULT_TYPE: "IMPALA",
+            VDK_IMPALA_HOST: "localhost",
+            VDK_IMPALA_PORT: "21050",
+        },
+    )
+    def test_lineage_not_collected_for_invalid_query(self):
+        mock_lineage_logger = mock.MagicMock(ILineageLogger)
+        runner = CliEntryBasedTestRunner(
+            TestConfigPlugin(mock_lineage_logger), impala_plugin
+        )
+        result: Result = runner.invoke(
+            ["run", jobs_path_from_caller_directory("sql-job-syntax-error")]
+        )
+        cli_assert_equal(1, result)
+        assert not mock_lineage_logger.send.called
+
+    @patch.dict(
+        os.environ,
+        {
+            VDK_DB_DEFAULT_TYPE: "IMPALA",
+            VDK_IMPALA_HOST: "localhost",
+            VDK_IMPALA_PORT: "21050",
+        },
+    )
+    def test_lineage_for_create_and_insert_queries(self):
+        mock_lineage_logger = mock.MagicMock(ILineageLogger)
+        runner = CliEntryBasedTestRunner(
+            TestConfigPlugin(mock_lineage_logger), impala_plugin
+        )
+        result: Result = runner.invoke(
+            ["run", jobs_path_from_caller_directory("sql-job")]
+        )
+        cli_assert_equal(0, result)
+
+        # the query is different on every run of the test
```

Comment:
don't quite understand this comment, what about the query is different on every run?

Reply:
every query begins with a comment that contains the op id of the job which is different for every execution, hence not validating the full query.